### ISSUE TITLE
Class "Google\ApiCore\Serializer" not found

### ISSUE THREAD

**Issue by NOaMTL** [kreait/firebase-php#676]
### Describe the issue you are experiencing

Hello,

I am having troubles to use firestore.
I get this error : 
`Unable to create a FirestoreClient: Class "Google\ApiCore\Serializer" not found`

my composer.json : 
```
{
    "type": "project",
    "license": "proprietary",
    "minimum-stability": "stable",
    "prefer-stable": true,
    "require": {
        "php": ">=8.0",
        "ext-ctype": "*",
        "ext-iconv": "*",
        "appaydin/pd-menu": "^3.0",
        "aws/aws-sdk-php": "3.0",
        "babdev/pagerfanta-bundle": "^3.3",
        "bukashk0zzz/liip-imagine-serialization-bundle": "^3.0",
        "composer/package-versions-deprecated": "1.11.99.2",
        "doctrine/doctrine-bundle": "^2.4",
        "doctrine/doctrine-migrations-bundle": "^3.1",
        "doctrine/orm": "^2.9",
        "easycorp/easyadmin-bundle": "^3.5",
        "fresh/vich-uploader-serialization-bundle": "^3.1",
        "friendsofsymfony/ckeditor-bundle": "^2.3",
        "gedmo/doctrine-extensions": "^3.1",
        "gesdinet/jwt-refresh-token-bundle": "^0.12.0",
        "google/cloud-firestore": "^0.1.0",
        "helios-ag/fm-elfinder-bundle": "^10.1",
        "jms/serializer-bundle": "^3.9",
        "knplabs/knp-gaufrette-bundle": "^0.7.1",
        "knplabs/knp-paginator-bundle": "^5.6",
        "kreait/firebase-bundle": "3.1.0",
        "lexik/jwt-authentication-bundle": "^2.12",
        "liip/imagine-bundle": "^2.6",
        "mhujer/jms-serializer-uuid-bundle": "^4.2",
        "nelmio/cors-bundle": "^2.2",
        "nesbot/carbon": "^2.50",
        "ramsey/uuid-doctrine": "^1.6",
        "sensio/framework-extra-bundle": "^6.1",
        "stof/doctrine-extensions-bundle": "^1.6",
        "symfony/console": "5.4.*",
        "symfony/dotenv": "5.4.*",
        "symfony/flex": "^1.3.1",
        "symfony/framework-bundle": "5.3.13",
        "symfony/monolog-bundle": "^3.7",
        "symfony/proxy-manager-bridge": "5.4.*",
        "symfony/rate-limiter": "5.4.*",
        "symfony/runtime": "5.4.*",
        "symfony/twig-bundle": "^5.3",
        "symfony/webpack-encore-bundle": "^1.12",
        "symfony/yaml": "5.4.*",
        "twig/extra-bundle": "^2.12|^3.0",
        "twig/intl-extra": "^3.3",
        "twig/twig": "^2.12|^3.0",
        "vich/uploader-bundle": "^1.17",
        "yidas/google-maps-services": "^1.0",
        "grpc/grpc": "~1.30.0"
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": {
            "*": "dist"
        },
        "sort-packages": true
    },
    "autoload": {
        "psr-4": {
            "App\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "App\\Tests\\": "tests/"
        }
    },
    "replace": {
        "symfony/polyfill-ctype": "*",
        "symfony/polyfill-iconv": "*",
        "symfony/polyfill-php72": "*"
    },
    "scripts": {
        "auto-scripts": {
            "cache:clear": "symfony-cmd",
            "assets:install %PUBLIC_DIR%": "symfony-cmd",
            "ckeditor:install": "symfony-cmd",
            "elfinder:install": "symfony-cmd"
        },
        "post-install-cmd": [
            "@auto-scripts"
        ],
        "post-update-cmd": [
            "@auto-scripts"
        ]
    },
    "conflict": {
        "symfony/symfony": "*"
    },
    "extra": {
        "symfony": {
            "allow-contrib": "true",
            "require": "5.4.*"
        }
    },
    "require-dev": {
        "doctrine/doctrine-fixtures-bundle": "^3.4",
        "symfony/maker-bundle": "^1.33",
        "symfony/stopwatch": "^5.4",
        "symfony/web-profiler-bundle": "^5.4"
    }
}
```

### Installed packages

```
appaydin/pd-menu                              3.0.02    Symfony Simple Menu Bundle
aws/aws-sdk-php                               3.0.0     AWS SDK for PHP - Use Amazon Web Services in your PHP project
babdev/pagerfanta-bundle                      v3.6.1    Bundle integrating Pagerfanta with Symfony
behat/transliterator                          v1.3.0    String transliterator
bukashk0zzz/liip-imagine-serialization-bundle v3.1.1    Provides integration between LiipImagineBundle and JMSSerializerBundle.
composer/package-versions-deprecated          1.11.99.2 Composer plugin that provides efficient querying for installed package versions (no runtime IO)
doctrine/annotations                          1.13.2    Docblock Annotations Parser
doctrine/cache                                1.12.1    PHP Doctrine Cache library is a popular cache implementation that supports many different drivers such as redis, memcache, apc, mongodb and others.
doctrine/collections                          1.6.8     PHP Doctrine Collections library that adds additional functionality on top of PHP arrays.
doctrine/common                               3.2.1     PHP Doctrine Common project is a library that provides additional functionality that other Doctrine projects depend on such as better reflection support, proxies and much...
doctrine/data-fixtures                        1.5.1     Data Fixtures for all Doctrine Object Managers
doctrine/dbal                                 3.2.0     Powerful PHP database abstraction layer (DBAL) with many features for database schema introspection and management.
doctrine/deprecations                         v0.5.3    A small layer on top of trigger_error(E_USER_DEPRECATED) or PSR-3 logging with options to disable all deprecations or selectively for packages.
doctrine/doctrine-bundle                      2.5.4     Symfony DoctrineBundle
doctrine/doctrine-fixtures-bundle             3.4.1     Symfony DoctrineFixturesBundle
doctrine/doctrine-migrations-bundle           3.2.1     Symfony DoctrineMigrationsBundle
doctrine/event-manager                        1.1.1     The Doctrine Event Manager is a simple PHP event system that was built to be used with the various Doctrine projects.
doctrine/inflector                            2.0.4     PHP Doctrine Inflector is a small library that can perform string manipulations with regard to upper/lowercase and singular/plural forms of words.
doctrine/instantiator                         1.4.0     A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                                1.2.1     PHP Doctrine Lexer parser library that can be used in Top-Down, Recursive Descent Parsers.
doctrine/migrations                           3.3.2     PHP Doctrine Migrations project offer additional functionality on top of the database abstraction layer (DBAL) for versioning your database schema and easily deploying ch...
doctrine/orm                                  2.10.4    Object-Relational-Mapper for PHP
doctrine/persistence                          2.2.3     The Doctrine Persistence project is a set of shared interfaces and functionality that the different Doctrine object mappers share.
doctrine/sql-formatter                        1.1.2     a PHP SQL highlighting library
easycorp/easyadmin-bundle                     v3.5.17   Admin generator for Symfony applications
fig/http-message-util                         1.1.5     Utility classes and constants for use with PSR-7 (psr/http-message)
firebase/php-jwt                              v5.5.1    A simple library to encode and decode JSON Web Tokens (JWT) in PHP. Should conform to the current spec.
fresh/vich-uploader-serialization-bundle      v3.1.2    Provides integration between VichUploaderBundle and JMSSerializerBundle.
friendsofphp/proxy-manager-lts                v1.0.5    Adding support for a wider range of PHP versions to ocramius/proxy-manager
friendsofsymfony/ckeditor-bundle              2.3.0     Provides a CKEditor integration for your Symfony project.
gedmo/doctrine-extensions                     v3.4.0    Doctrine2 behavioral extensions
gesdinet/jwt-refresh-token-bundle             v0.12.0   Implements a refresh token system over Json Web Tokens in Symfony
google/auth                                   v1.18.0   Google Auth Library for PHP
google/cloud-core                             v1.43.1   Google Cloud PHP shared dependency, providing functionality useful to all components.
google/cloud-firestore                        v0.1.0    Cloud Firestore client for PHP
google/cloud-storage                          v1.25.2   Cloud Storage Client for PHP
google/crc32                                  v0.1.0    Various CRC32 implementations
google/gax                                    0.25.0    Google API Extensions for PHP
google/proto-client                           0.25.0    Generated proto and gRPC classes for Google Cloud Platform in PHP
google/protobuf                               v3.19.1   proto library for PHP
grpc/grpc                                     1.42.0    gRPC library for PHP
guzzlehttp/guzzle                             6.5.5     Guzzle is a PHP HTTP client library
guzzlehttp/promises                           1.5.1     Guzzle promises library
guzzlehttp/psr7                               1.8.3     PSR-7 message implementation that also provides common utility methods
helios-ag/fm-elfinder-bundle                  10.1      ElFinder bundle, adds ElFinder file manager to your Symfony project
imagine/imagine                               1.2.4     Image processing for PHP 5.3
jms/metadata                                  2.6.1     Class/method/property metadata management in PHP
jms/serializer                                3.17.1    Library for (de-)serializing data of any complexity; supports XML, JSON, and YAML.
jms/serializer-bundle                         3.10.0    Allows you to easily serialize, and deserialize data of any complexity
knplabs/gaufrette                             v0.9.0    PHP library that provides a filesystem abstraction layer
knplabs/knp-components                        v3.5.0    Knplabs component library
knplabs/knp-gaufrette-bundle                  v0.7.1    Allows to easily use the Gaufrette library in a Symfony project
knplabs/knp-paginator-bundle                  v5.8.0    Paginator bundle for Symfony to automate pagination and simplify sorting and other features
kreait/clock                                  1.1.0     A PHP 7.0 compatible clock abstraction
kreait/firebase-bundle                        3.1.0     Symfony Bundle for the Firebase Admin SDK
kreait/firebase-php                           5.25.0    Firebase Admin SDK
kreait/firebase-tokens                        1.16.1    A library to work with Firebase tokens
laminas/laminas-code                          4.5.1     Extensions to the PHP Reflection API, static code scanning, and code generation
lcobucci/clock                                2.1.0     Yet another clock abstraction
lcobucci/jwt                                  4.0.4     A simple library to work with JSON Web Token and JSON Web Signature
lexik/jwt-authentication-bundle               v2.14.3   This bundle provides JWT authentication for your Symfony REST API
liip/imagine-bundle                           2.7.4     This bundle provides an image manipulation abstraction toolkit for Symfony-based projects.
mhujer/jms-serializer-uuid                    3.4.1     Uuid serializer and deserializer for JMS Serializer library
mhujer/jms-serializer-uuid-bundle             4.2.0     Symfony integration for Uuid serializer and deserializer for JMS Serializer library
monolog/monolog                               2.3.5     Sends your logs to files, sockets, inboxes, databases and various web services
mtdowling/jmespath.php                        2.6.1     Declaratively specify how to extract elements from a JSON document
namshi/jose                                   7.2.3     JSON Object Signing and Encryption library for PHP.
nelmio/cors-bundle                            2.2.0     Adds CORS (Cross-Origin Resource Sharing) headers support in your Symfony application
nesbot/carbon                                 2.55.2    An API extension for DateTime that supports 281 different languages.
nikic/php-parser                              v4.13.2   A PHP parser written in PHP
pagerfanta/core                               v3.5.1    Core Pagerfanta API
paragonie/random_compat                       v9.99.100 PHP 5.x polyfill for random_bytes() and random_int() from PHP 7
phpstan/phpdoc-parser                         1.2.0     PHPDoc parser with support for nullable, intersection and generic types
psr/cache                                     2.0.0     Common interface for caching libraries
psr/container                                 1.1.2     Common Container Interface (PHP FIG PSR-11)
psr/event-dispatcher                          1.0.0     Standard interfaces for event handling.
psr/http-message                              1.0.1     Common interface for HTTP messages
psr/log                                       2.0.0     Common interface for logging libraries
psr/simple-cache                              1.0.1     Common interfaces for simple caching
ralouphie/getallheaders                       3.0.3     A polyfill for getallheaders.
ramsey/uuid                                   3.9.6     Formerly rhumsaa/uuid. A PHP 5.4+ library for generating RFC 4122 version 1, 3, 4, and 5 universally unique identifiers (UUID).
ramsey/uuid-doctrine                          1.8.0     Use ramsey/uuid as a Doctrine field type.
riverline/multipart-parser                    2.0.9     One class library to parse multipart content with encoding and charset support.
rize/uri-template                             0.3.4     PHP URI Template (RFC 6570) supports both expansion & extraction
sensio/framework-extra-bundle                 v6.2.4    This bundle provides a way to configure your controllers with annotations
stof/doctrine-extensions-bundle               v1.7.0    Integration of the gedmo/doctrine-extensions with Symfony
studio-42/elfinder                            2.1.60    File manager for web
symfony/asset                                 v5.4.0    Manages URL generation and versioning of web assets such as CSS stylesheets, JavaScript files and image files
symfony/cache                                 v5.4.2    Provides an extended PSR-6, PSR-16 (and tags) implementation
symfony/cache-contracts                       v2.5.0    Generic abstractions related to caching
symfony/config                                v5.4.2    Helps you find, load, combine, autofill and validate configuration values of any kind
symfony/console                               v5.4.2    Eases the creation of beautiful and testable command line interfaces
symfony/dependency-injection                  v5.4.2    Allows you to standardize and centralize the way objects are constructed in your application
symfony/deprecation-contracts                 v2.5.0    A generic function and convention to trigger deprecation notices
symfony/doctrine-bridge                       v5.4.2    Provides integration for Doctrine with various Symfony components
symfony/dotenv                                v5.4.2    Registers environment variables from a .env file
symfony/error-handler                         v5.4.2    Provides tools to manage errors and ease debugging PHP code
symfony/event-dispatcher                      v5.4.0    Provides tools that allow your application components to communicate with each other by dispatching events and listening to them
symfony/event-dispatcher-contracts            v3.0.0    Generic abstractions related to dispatching event
symfony/expression-language                   v5.4.0    Provides an engine that can compile and evaluate expressions
symfony/filesystem                            v5.4.0    Provides basic utilities for the filesystem
symfony/finder                                v5.4.2    Finds files and directories via an intuitive fluent interface
symfony/flex                                  v1.17.6   Composer plugin for Symfony
symfony/form                                  v5.4.2    Allows to easily create, process and reuse HTML forms
symfony/framework-bundle                      v5.3.13   Provides a tight integration between Symfony components and the Symfony full-stack framework
symfony/http-foundation                       v5.4.2    Defines an object-oriented layer for the HTTP specification
symfony/http-kernel                           v5.4.2    Provides a structured process for converting a Request into a Response
symfony/intl                                  v5.4.2    Provides a PHP replacement layer for the C intl extension that includes additional data from the ICU library
symfony/lock                                  v5.4.2    Creates and manages locks, a mechanism to provide exclusive access to a shared resource
symfony/maker-bundle                          v1.36.4   Symfony Maker helps you create empty commands, controllers, form classes, tests and more so you can forget about writing boilerplate code.
symfony/mime                                  v5.4.2    Allows manipulating MIME messages
symfony/monolog-bridge                        v5.4.0    Provides integration for Monolog with various Symfony components
symfony/monolog-bundle                        v3.7.1    Symfony MonologBundle
symfony/options-resolver                      v5.4.0    Provides an improved replacement for the array_replace PHP function
symfony/password-hasher                       v5.4.2    Provides password hashing utilities
symfony/polyfill-intl-grapheme                v1.23.1   Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-icu                     v1.23.0   Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn                     v1.23.0   Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer              v1.23.0   Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring                     v1.23.1   Symfony polyfill for the Mbstring extension
symfony/polyfill-php56                        v1.20.0   Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-php73                        v1.23.0   Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/polyfill-php80                        v1.23.1   Symfony polyfill backporting some PHP 8.0+ features to lower PHP versions
symfony/polyfill-php81                        v1.23.0   Symfony polyfill backporting some PHP 8.1+ features to lower PHP versions
symfony/polyfill-uuid                         v1.23.0   Symfony polyfill for uuid functions
symfony/process                               v5.4.2    Executes commands in sub-processes
symfony/property-access                       v5.4.2    Provides functions to read and write from/to an object or array using a simple string notation
symfony/property-info                         v5.4.2    Extracts information about PHP class' properties using metadata of popular sources
symfony/proxy-manager-bridge                  v5.4.2    Provides integration for ProxyManager with various Symfony components
symfony/rate-limiter                          v5.4.2    Provides a Token Bucket implementation to rate limit input and output in your application
symfony/routing                               v5.4.0    Maps an HTTP request to a set of configuration variables
symfony/runtime                               v5.4.1    Enables decoupling PHP applications from global state
symfony/security-bundle                       v5.4.2    Provides a tight integration of the Security component into the Symfony full-stack framework
symfony/security-core                         v5.4.2    Symfony Security Component - Core Library
symfony/security-csrf                         v5.4.0    Symfony Security Component - CSRF Library
symfony/security-guard                        v5.4.0    Symfony Security Component - Guard
symfony/security-http                         v5.4.2    Symfony Security Component - HTTP Integration
symfony/service-contracts                     v2.5.0    Generic abstractions related to writing services
symfony/stopwatch                             v5.4.0    Provides a way to profile code
symfony/string                                v5.4.2    Provides an object-oriented API to strings and deals with bytes, UTF-8 code points and grapheme clusters in a unified way
symfony/translation                           v5.4.2    Provides tools to internationalize your application
symfony/translation-contracts                 v2.5.0    Generic abstractions related to translation
symfony/twig-bridge                           v5.4.0    Provides integration for Twig with various Symfony components
symfony/twig-bundle                           v5.4.0    Provides a tight integration of Twig into the Symfony full-stack framework
symfony/uid                                   v5.4.2    Provides an object-oriented API to generate and represent UIDs
symfony/validator                             v5.4.2    Provides tools to validate values
symfony/var-dumper                            v5.4.2    Provides mechanisms for walking through any arbitrary PHP variable
symfony/var-exporter                          v5.4.2    Allows exporting any serializable PHP data structure to plain PHP code
symfony/web-profiler-bundle                   v5.4.2    Provides a development tool that gives detailed information about the execution of any request
symfony/webpack-encore-bundle                 v1.13.2   Integration with your Symfony app & Webpack Encore!
symfony/yaml                                  v5.4.2    Loads and dumps YAML files
twig/extra-bundle                             v3.3.4    A Symfony bundle for extra Twig extensions
twig/intl-extra                               v3.3.4    A Twig extension for Intl
twig/twig                                     v3.3.4    Twig, the flexible, fast, and secure template language for PHP
vich/uploader-bundle                          1.19.0    Ease file uploads attached to entities
yidas/google-maps-services                    1.0.3     PHP client library(SDK) for Google Maps API Web Services
```

### PHP version and extensions

OS : Debian 9.13 (Plesk)

```
composer-plugin-api    2.1.0    The Composer Plugin API
composer-runtime-api   2.1.0    The Composer Runtime API
ext-bcmath             8.0.9    The bcmath PHP extension
ext-bz2                8.0.9    The bz2 PHP extension
ext-calendar           8.0.9    The calendar PHP extension
ext-ctype              8.0.9    The ctype PHP extension
ext-curl               8.0.9    The curl PHP extension
ext-date               8.0.9    The date PHP extension
ext-dom                20031129 The dom PHP extension
ext-exif               8.0.9    The exif PHP extension
ext-fileinfo           8.0.9    The fileinfo PHP extension
ext-filter             8.0.9    The filter PHP extension
ext-gd                 8.0.9    The gd PHP extension
ext-gettext            8.0.9    The gettext PHP extension
ext-gmp                8.0.9    The gmp PHP extension
ext-hash               8.0.9    The hash PHP extension
ext-iconv              8.0.9    The iconv PHP extension
ext-imap               8.0.9    The imap PHP extension
ext-intl               8.0.9    The intl PHP extension
ext-json               8.0.9    The json PHP extension
ext-ldap               8.0.9    The ldap PHP extension
ext-libxml             8.0.9    The libxml PHP extension
ext-mbstring           8.0.9    The mbstring PHP extension
ext-mysqli             8.0.9    The mysqli PHP extension
ext-mysqlnd            0        The mysqlnd PHP extension (actual version: mysqlnd 8.0.9)
ext-openssl            8.0.9    The openssl PHP extension
ext-pcre               8.0.9    The pcre PHP extension
ext-pdo                8.0.9    The PDO PHP extension
ext-pdo_mysql          8.0.9    The pdo_mysql PHP extension
ext-pdo_sqlite         8.0.9    The pdo_sqlite PHP extension
ext-phar               8.0.9    The Phar PHP extension
ext-readline           8.0.9    The readline PHP extension
ext-reflection         8.0.9    The Reflection PHP extension
ext-session            8.0.9    The session PHP extension
ext-simplexml          8.0.9    The SimpleXML PHP extension
ext-soap               8.0.9    The soap PHP extension
ext-sockets            8.0.9    The sockets PHP extension
ext-spl                8.0.9    The SPL PHP extension
ext-tokenizer          8.0.9    The tokenizer PHP extension
ext-xml                8.0.9    The xml PHP extension
ext-xmlreader          8.0.9    The xmlreader PHP extension
ext-xmlwriter          8.0.9    The xmlwriter PHP extension
ext-xsl                8.0.9    The xsl PHP extension
ext-zip                1.19.3   The zip PHP extension
ext-zlib               8.0.9    The zlib PHP extension
lib-bz2                1.0.8    The bz2 library
lib-curl               7.76.1   The curl library
lib-curl-libssh2       1.9.0    curl libssh2 version
lib-curl-openssl       1.1.1.11 curl OpenSSL version (1.1.1.11)
lib-curl-zlib          1.2.11   curl zlib version
lib-date-timelib       2020.02  date timelib version
lib-date-zoneinfo      2021.1   zoneinfo ("Olson") database for date
lib-fileinfo-libmagic  539      fileinfo libmagic version
lib-gd                 2.0.35   The gd library
lib-gd-freetype        2.9.1    freetype version for gd
lib-gd-libjpeg         9.0      libjpeg version for gd
lib-gd-libpng          1.6.34   libpng version for gd
lib-gd-libxpm          3.5.12   libxpm version for gd
lib-gmp                6.0.0    The gmp library
lib-iconv              1.16     The iconv library
lib-icu                68.2     The ICU unicode and globalization support library
lib-icu-cldr           38.1     ICU CLDR project version
lib-icu-unicode        13.0.0   ICU unicode version
lib-ldap-openldap      2.4.47   OpenLDAP version of ldap
lib-libxml             2.9.10   libxml library version
lib-libxslt            1.1.34   The libxslt library
lib-libxslt-libxml     2.9.10   libxml version libxslt is compiled against
lib-mbstring-libmbfl   1.3.2    mbstring libmbfl version
lib-mbstring-oniguruma 6.9.7    mbstring oniguruma version
lib-openssl            1.1.1.11 OpenSSL 1.1.1k  25 Mar 2021
lib-pcre               10.35    The pcre library
lib-pcre-unicode       13.0.0   PCRE Unicode version support
lib-pdo_sqlite-sqlite  3.33.0   The pdo_sqlite-sqlite library
lib-zip-libzip         1.7.1    The zip-libzip library
lib-zlib               1.2.11   The zlib library
php                    8.0.9    The PHP interpreter
php-64bit              8.0.9    The PHP interpreter, 64bit
php-ipv6               8.0.9    The PHP interpreter, with IPv6 support
php-zts                8.0.9    The PHP interpreter, with Zend Thread Safety
```

### On which operating system(s) does the issue occur?

- [X] Linux
- [ ] MacOS
- [ ] Windows

### Steps to reproduce the issue.

My controller  injecting (Firestore $firestore): 

```
 $db = $firestore->database();
        $data = [
            'name' => 'Los Angeles',
            'state' => 'CA',
            'country' => 'USA'
        ];
        $db->collection('test-collection')->document('test')->set($data);
```

### Error message/Stack trace

Unable to create a FirestoreClient: Class "Google\ApiCore\Serializer" not found

`     return $this->createFactory($config)->createDatabase();    }    public function createFirestore(array $config = []): Firebase\Contract\Firestore    {        return $this->createFactory($config)->createFirestore();    }    public function createMessaging(array $config = []): Firebase\Contract\Messaging    {        return $this->createFactory($config)->createMessaging();`

**Comment by jeromegamez** [comment]
You have a lot of dependencies that could interfere with each other, so that one could prevent the installation of a recent version of another. Here are some issues that stood out to me:

`google/cloud-firestore` is set to `^0.1.0` - the current version is 1.20.0, which you will not get with this constraint.

`kreait/firebase-bundle` is fixed to `3.1.0` - this will prevent you from getting newer releases, you should set it to `^3.1.0`.

You can check other outdated dependencies with the `composer outdated` command.

**Comment by jeromegamez** [comment]
As this is not an issue with the SDK itself but with composer dependency management, I'm going to close this issue, but feel free to share the steps you did to update the dependencies so that others might benefit from them.
